src/main/java/duke/storage/ListLoader.java

```diff
@@ -0,0 +1,281 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import duke.exception.DukeException;
+import duke.list.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskId;
+import duke.task.Todo;
+
+/**
+ * Reads and writes to a skeleton of the current task list while being updated as the list changes.
+ *
+ * @author WR3nd3
+ */
+public class ListLoader {
+    /** File to read and write duke.list.TaskList tasks to */
+    private File listText = new File("data/duke.txt");
+
+    /** duke.list.TaskList to update */
+    private TaskList taskList;
+
+
+    /**
+     * Constructor for duke.storage.ListLoader.
+     *
+     * @param taskList The duke.list.TaskList to be updated with the stored data.
+     */
+    public ListLoader(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Reads stored data in the saved list and updates the task list.
+     *
+     * @throws DukeException for failing to load the file.
+     */
+    public void load() throws DukeException {
+        BufferedReader reader = null;
+        try {
+            // Create necessary save file if it does not exist
+            if (!new File("data").exists()) {
```

Comment:
Maybe just create the File object once (before the if statement) and then do the checks instead of two separate objects?